docs/extensibility/payment-method-integration.md

```diff
@@ -0,0 +1,131 @@
+- `edit` (required): This should be a react node that will be output in the express payment method area when the block is rendered in the editor. It will be cloned in the rendering process. When cloned, this react node will receive props from the payment method interface to checkout (but they will contain preview data). 
+- `canMakePayment` (required): This should be a function that returns whether the payment method can make a payment in the current environment. The function _can_ return a promise and will receive [`cartData`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/308e968c700028180cab391f2223eb0a43dd2d4d/assets/js/type-defs/cart.js#L163-L177) as an argument (so the payment method can use that to determine if it can be used). Most payment methods will probably just return `true` from the function, but this does allow payment methods like `ApplePay` which have restrictions on where they work to return `false` and thus be excluded as an option in specific browser environments (or other conditions). Keep in mind this function is only invoked once in the mounting process of the checkout block. 
+- `paymentMethodId`: This is the only optional configuration object. The value of this property is what will accompany the checkout processing request to the server and used to identify what payment method gateway class to load for processing the payment (if the shopper selected the gateway). So for instance if this is `stripe`, then `WC_Gateway_Stripe::process_payment` will be invoked for processing the payment.
+
+### Payment Methods - `registerPaymentMethod( paymentMethodCreator )`
+
+![Payment Method Area](https://user-images.githubusercontent.com/1429108/79565774-5d230700-807f-11ea-9335-0111ec306a47.png)
+
+To register a payment method, you use the `registerPaymentMethod` function from the blocks registry. An example of importing this for use in your JavaScript file is:
+
+_Aliased import_
+```js
+import { registerPaymentMethod } from '@woocommerce/blocks-registry';
+```
+
+_wc global_
+```js
+const { registerPaymentMethod } = wc.wcBlocksRegistry;
+```
+
+The registry function expects a function that will receive an `PaymentMethodConfig` creator as an argument and is expected to return a valid `PaymentMethodConfig` instance.  As a very basic example, something like this:
+
+```js
+registerPaymentMethod( ( PaymentMethodConfig ) => new PaymentMethodConfig( options ) );
+```
+
+The options you feed the configuration instance are the same as those for express payment methods with the following additions:
+
+- `label`: This should be a react node that will be used to output the label for the tab in the payment methods are. For example it might be `<strong>Credit/Debit Cart</strong>` or you might output images.
+- `ariaLabel`: This is the label that will be read out via screen-readers when the payment method is selected.
+
+### Props fed to payment method nodes
```

Comment:
Pedantic but we're mixing title case with lowercase headings :) Lets use title case.